### PR TITLE
BAEL-8000: Conditional processing using the Stream API

### PR DESCRIPTION
**Pull request by azhwani**
+ Add test cases to highlight how to use **_Collectors.partitioningBy_**, _**Collectors.groupingBy**_, and **custom consumer** to implement if/else Logic in Java 8 Streams
+ Keep the same existing assertion structure/logic (Junit 4)

### CODE REVIEW

File: core-java-modules/core-java-streams-3/src/test/java/com/baeldung/streams/conditional/StreamForEachIfElseUnitTest.java [hunk 1]
```diff
@@ -24,6 +27,22 @@ public final void givenIntegerStream_whenCheckingIntegerParityWithIfElse_thenEns
...
+    public final void givenIntegerStream_whenCheckingIntegerParityWithCustomConsumer_thenEnsureCorrectParity() {
+        List<Integer> ints = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
+
+        Consumer<Integer> integerParityAsserter = i -> {
+            if (i.intValue() % 2 == 0) {
+                Assert.assertTrue(i.intValue() + " is not even", i.intValue() % 2 == 0);
+            } else {
+                Assert.assertTrue(i.intValue() + " is not odd", i.intValue() % 2 != 0);
+            }
+        };
```
Comment (JavierPeris) [inline core-java-modules/core-java-streams-3/src/test/java/com/baeldung/streams/conditional/StreamForEachIfElseUnitTest.java:40]: We can do without the unboxing:
```suggestion
        Consumer<Integer> integerParityAsserter = i -> {
            if (i % 2 == 0) {
                Assert.assertTrue(i + " is not even", i % 2 == 0);
            } else {
                Assert.assertTrue(i + " is not odd", i % 2 != 0);
            }
        };
```

File: core-java-modules/core-java-streams-3/src/test/java/com/baeldung/streams/conditional/StreamForEachIfElseUnitTest.java [hunk 2]
```diff
@@ -38,4 +57,26 @@ public final void givenIntegerStream_whenCheckingIntegerParityWithStreamFilter_t
 
     }
 
+    @Test
+    public final void givenIntegerStream_whenCheckingIntegerParityWithStreamPartitioningBy_thenEnsureCorrectParity() {
+        List<Integer> ints = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
+
+        Map<Boolean, List<Integer>> resultMap = ints.stream()
+            .collect(Collectors.partitioningBy(i -> i.intValue() % 2 == 0));
```
Comment (JavierPeris) [inline core-java-modules/core-java-streams-3/src/test/java/com/baeldung/streams/conditional/StreamForEachIfElseUnitTest.java:65]: Same here.

File: core-java-modules/core-java-streams-3/src/test/java/com/baeldung/streams/conditional/StreamForEachIfElseUnitTest.java [hunk 3]
```diff
@@ -38,4 +57,26 @@ public final void givenIntegerStream_whenCheckingIntegerParityWithStreamFilter_t
...
+        Assert.assertEquals(Arrays.asList(2, 4, 6, 8, 10), resultMap.get(true));
+        Assert.assertEquals(Arrays.asList(1, 3, 5, 7, 9), resultMap.get(false));
+    }
+
+    @Test
+    public final void givenIntegerStream_whenCheckingIntegerParityWithStreamGroupingBy_thenEnsureCorrectParity() {
+        List<Integer> ints = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
+
+        Map<String, List<Integer>> resultMap = ints.stream()
+            .collect(Collectors.groupingBy(i -> i.intValue() % 2 == 0 ? "even" : "odd"));
```
Comment (JavierPeris) [inline core-java-modules/core-java-streams-3/src/test/java/com/baeldung/streams/conditional/StreamForEachIfElseUnitTest.java:76]: And here.

File: core-java-modules/core-java-streams-3/src/test/java/com/baeldung/streams/conditional/StreamForEachIfElseUnitTest.java (hunk 1 above)
Comment (azhwani) [reply core-java-modules/core-java-streams-3/src/test/java/com/baeldung/streams/conditional/StreamForEachIfElseUnitTest.java:40]: Yes, the idea was to keep the same logic as the other test cases, but yeah I will remove the intValue() method

Comment (azhwani) [reply core-java-modules/core-java-streams-3/src/test/java/com/baeldung/streams/conditional/StreamForEachIfElseUnitTest.java]: Done

File: core-java-modules/core-java-streams-3/src/test/java/com/baeldung/streams/conditional/StreamForEachIfElseUnitTest.java (hunk 2 above)
Comment (azhwani) [reply core-java-modules/core-java-streams-3/src/test/java/com/baeldung/streams/conditional/StreamForEachIfElseUnitTest.java]: Done

File: core-java-modules/core-java-streams-3/src/test/java/com/baeldung/streams/conditional/StreamForEachIfElseUnitTest.java (hunk 3 above)
Comment (azhwani) [reply core-java-modules/core-java-streams-3/src/test/java/com/baeldung/streams/conditional/StreamForEachIfElseUnitTest.java]: Done

**Pull request merged**